### ISSUE TITLE
BCDB: could not find obj with id  

### ISSUE THREAD

**Issue by xmonader** [threefoldtech/jumpscaleX_core#170]
on fresh containers and inconsistenly happening after pulling (packages locations updates .. etc)
![image](https://user-images.githubusercontent.com/64129/68466983-169ec600-021e-11ea-8ee4-6dbe26b54e64.png)

- index should get rebuilt automatically in that case
maybe relevant: https://github.com/threefoldtech/jumpscaleX_core/issues/168

**Comment by abom** [comment]
Maybe related error, when starting threebot server

```
Sun 10 12:04:39 c28b62fa9a5cf998c.py -  31 - b_ede010359363515c28b62fa9a5cf998c : init index:jumpscale.myjobs.job                    
Sun 10 12:04:39 129577613d9be1f27.py -  31 - n_2c467e8e2116e34129577613d9be1f27 : init index:jumpscale.myjobs.action                 
Sun 10 12:04:40 ta/bcdb/BCDBModel.py - 503 - get                                : EXCEPTION:                                         
    no object with id 1 found in model:jumpscale.myjobs.action
    , this means the index gave back an id which is not part of this model.
--TRACEBACK------------------
/sandbox/var/cmds/threebot_default.py in <module>
    5    server.start(background=False)
/sandbox/lib/jumpscale/Jumpscale/servers/threebot/ThreebotServer.py in start
    177    package.start()
/sandbox/lib/jumpscale/Jumpscale/tools/threebot_package/ThreeBotPackage.py in start
    124    self._init_before_action()
/sandbox/lib/jumpscale/Jumpscale/tools/threebot_package/ThreeBotPackage.py in _init_before_action
    78    j.servers.myjobs.schedule(load_wiki, wiki_name=name, wiki_path=path)
/sandbox/lib/jumpscale/Jumpscale/servers/myjobs/MyJobsFactory.py in schedule
    425    name=name, method=method, kwargs=kwargs, dependencies=dependencies, args_replace=args_replace
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSConfigsBCDB.py in new
    44    jsconfig = self._new(name=name, jsxobject=jsxobject, autosave=autosave, **kwargs)
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSConfigsBCDB.py in _new
    93    jsconfig = jsconfig_klass(parent=self, jsxobject=jsxobject, **kwargs_to_class)
/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSBase.py in __init__
    63    self._init(**kwargs)
/sandbox/lib/jumpscale/Jumpscale/servers/myjobs/MyJobs.py in _init
    27    self.process_code(method, args_replace)
/sandbox/lib/jumpscale/Jumpscale/servers/myjobs/MyJobs.py in process_code
    46    new, action = j.servers.myjobs.action_get(key)
/sandbox/lib/jumpscale/Jumpscale/servers/myjobs/MyJobsFactory.py in action_get
    60    res = self.model_action.find(key=key)
/sandbox/lib/jumpscale/Jumpscale/data/bcdb/BCDBModel.py in find
    628    res.append(self.get(id))
/sandbox/lib/jumpscale/Jumpscale/data/bcdb/BCDBDecorator.py in wrapper_queue_method
    60    res = func(*args, **kwargs)
/sandbox/lib/jumpscale/Jumpscale/data/bcdb/BCDBModel.py in get
    503    obj_id, self
-----------------------------

Sun 10 12:04:41 BCDB.py              - 338 - bcdb                               : DATAPROCESSOR & SQLITE STOPPED OK
Sun 10 12:04:42 BCDB.py              - 338 - bcdb                               : DATAPROCESSOR & SQLITE STOPPED OK
```

**Comment by xmonader** [comment]
should be fixed in 4409a78552b0915ff671c5217c08e280ffabd0d4 (we never return an id from bcdb config without explicitly stating it should exist on the model) there should be some rethinking of the current design.

**Comment by Pishoy** [comment]
as per this ticket https://github.com/threefoldtech/jumpscaleX_core/issues/168
i did not face this issue again
